libs/@blockprotocol/graph/src/types/temporal-versioning.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Types used in embedding applications and blocks that support multi-axis temporal versioning schemes.
+ */
+
+import { Subtype } from "../util.js";
+
+/**
+ * An ISO 8601 formatted timestamp string
+ */
+export type Timestamp = string;
+
+/**
+ * @todo - doc
+ */
+export type TemporalAxes = "transactionTime" | "decisionTime";
+
+/**
+ * The bound of a time-interval that is either exclusively or inclusively limited by a `Timestamp`
+ */
+export type TimestampLimitedTemporalBound = {
```

Comment:
Do we need the word `Timestamp` in these types? Doesn't 'limited temporal bound' already imply it?

Reply:
This is another one of those things I had intended to open a comment highlighting but forgot to 😅. I'm really uncomfortable with how long and unwieldy some of these type names are, they really are pretty rough. That said, I opted for correctness and explicitness over something being accidentally misleading.

For this specific case, I'm not against us going to `LimitedTemporalBound`, I was just trying to say it had a timestamp _at_ its limit, (because a potential interpretation of "unbounded" is that its limit is infinity, or negative infinity)

Reply:
I think because we have an explicit separate type for `Unbounded`, we could get away with this being `LimitedTemporalBound`. Can think about it in follow up though

Reply:
Not hard to do, have pushed